vos/vos/vos.py

```diff
@@ -2732,6 +2781,78 @@ def delete(self, uri):
             response = self.get_session(uri).delete(url)
             response.raise_for_status()
 
+    def recursive_delete(self, uri):
+        """Delete the node and its content.
+        :param uri: The (Container/Link/Data)Node to delete from the service.
+
+        :returns: tuple of the form (successfull_deletes, failed_deletes)
+        :raises When a network problem occurs, it raises one of the
+        HttpException exceptions declared in the
+        cadcutils.exceptions module
+        """
+        uri = self.fix_uri(uri)
+        logger.debug("recursive delete {0}".format(uri))
+        with nodeCache.volatile(uri):
+            session = self.get_session(uri)
+            # quickly check target exists
+            self.get_node(uri)
+            response = session.post(self.get_endpoints(uri).recursive_del, {'target': uri}, allow_redirects=False)
+            response.raise_for_status()
+            if response.status_code != 303:
+                raise RuntimeError('Unexpected response for running job: '
+                                   + response.status_code)
+            return self._run_recursive_job(session, response.headers['location'])
+
+    def _run_recursive_job(self, session, url):
+        # runs an already created recursive job and returns the number of
+        # successfull and unsuccessfull actions
+        logger.debug('POST: ' + url)
+        response = session.post(url + '/phase', data={'phase': 'RUN'}, allow_redirects=False)
+        if response.status_code != 303:
+            raise RuntimeError('Unexpected response for running job: '
+                               + response.status_code)
+
+        # polling: WAIT will block for up to 6 sec or until phase change or if job is in
+        # a terminal phase
+        jobPoll = url + '?WAIT=6'
+        count = 0
+        done = False
+        while not done and count < 100:  # max 100*6 = 600 sec polling
```

Comment:
Use constant values for the `6` and the `100` to make it easier to over-ride for testing?

Reply:
I hope `vostools` will soon be re-factor at which point this and other similar style issues will be addressed (and tested).